File: 0144-binary-tree-preorder-traversal/0144-binary-tree-preorder-traversal.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func preorderTraversal(root *TreeNode) []int {
    data := make([]int, 0, 10)
	preorder(root, &data)
	return data
}

func preorder(root *TreeNode, data *[]int) {
	if root == nil {
		return
	}
    
	*data = append(*data, root.Val)
	
    preorder(root.Left, data)
	preorder(root.Right, data)
}

File: 0144-binary-tree-preorder-traversal/0144-binary-tree-preorder-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> preorderTraversal(TreeNode root) {
        var response = new ArrayList<Integer>();
        preorder(root, response);
        return response;
    }
    
    public void preorder(TreeNode root, ArrayList<Integer> response) {
        if(root == null)
            return;
        
        response.add(root.val);
        preorder(root.left, response);
        preorder(root.right, response);
        
    }
}